src/libraries/Common/src/Interop/Unix/System.Native/Interop.SocketAddress.cs

```diff
@@ -9,9 +9,9 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetIPSocketAddressSizes")]
+        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetSocketAddressSizes")]
         [SuppressGCTransition]
-        internal static unsafe partial Error GetIPSocketAddressSizes(int* ipv4SocketAddressSize, int* ipv6SocketAddressSize);
+        internal static partial Error GetSocketAddressSizes(ref int ipv4SocketAddressSize, ref int ipv6SocketAddressSize, ref int udsSocketAddressSize, ref int maxSocketAddressSize);
```

Comment:
LibraryImport is going to need to generate a stub for this, which will be a bit complicated due to needing to generate pinning for each of these refs.  If this were instead written with int* instead of ref int, no stub would be necessary at all I believe.  And it looks like it's just called from one place, where you could just write, e.g.:
```C#
int ipv4 = 0, ipv6 = 0, uds = 0, max = 0;
GetSocketAddressSizes(&ipv4, &ipv6, &uds, &max);
IPv4AddressSize = ipv4;
... // copy the others
```
It's not a hot path so not a big deal, though.

Reply:
ok. I can leave it as int*

Reply:
updated + I added some Asserts.